webrtc/RTCDtlsTransport-getRemoteCertificates.html

```diff
@@ -4,104 +4,102 @@
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="RTCPeerConnection-helper.js"></script>
+<script src="RTCTransport-helper.js"></script>
 <script>
   'use strict';
 
   // The following helper functions are called from RTCPeerConnection-helper.js:
-  //   exchangeIceCandidates
-  //   doSignalingHandshake
-
-  /*
-    5.5.  RTCDtlsTransport Interface
-      interface RTCDtlsTransport : EventTarget {
-        readonly attribute RTCDtlsTransportState state;
-        sequence<ArrayBuffer> getRemoteCertificates();
-                 attribute EventHandler          onstatechange;
-                 attribute EventHandler          onerror;
-        ...
-      };
-
-      enum RTCDtlsTransportState {
-        "new",
-        "connecting",
-        "connected",
-        "closed",
-        "failed"
-      };
-
-      getRemoteCertificates
-        Returns the certificate chain in use by the remote side, with each certificate
-        encoded in binary Distinguished Encoding Rules (DER) [X690].
-        getRemoteCertificates() will return an empty list prior to selection of the
-        remote certificate, which will be completed by the time RTCDtlsTransportState
-        transitions to "connected".
-   */
-  async_test(t => {
+  //  exchangeIceCandidates
+  //  doSignalingHandshake
+
+  // The following helper functions are called from RTCTransport-helper.js:
+  //  waitForConnectedState
+
+  function validateConnectedDtlsTransport(dtlsTransport) {
+    const certs = dtlsTransport.getRemoteCertificates();
+
+    assert_greater_than(certs.length, 0,
+      'Expect DTLS transport to have at least one remote certificate when connected');
+
+    for (const cert of certs) {
+      assert_true(cert instanceof ArrayBuffer,
+        'Certificates should be represented as ArrayBuffers');
+    }
+  }
+
+  function validateConnectingDtlsTransport(dtlsTransport) {
+    if (dtlsTransport.state !== 'connected') {
+      assert_array_equals(dtlsTransport.getCertificates(), [],
+        'The DTLS certificate list should be empty before the transport is connected');
+    }
+  }
+
+  promise_test(async t => {
+    const pc1 = new RTCPeerConnection();
+    t.add_cleanup(() => pc1.close());
+
+    const pc2 = new RTCPeerConnection();
+    t.add_cleanup(() => pc2.close());
+
+    const [track] = await getTrackFromUserMedia('audio');
+    pc1.addTransceiver(track);
+    exchangeIceCandidates(pc1, pc2);
+
+    await doSignalingHandshake(pc1, pc2);
+
+    const dtlsTransports = [
+      ...pc1.getSenders(),
+      ...pc1.getReceivers(),
+      ...pc2.getSenders(),
+      ...pc2.getReceivers()
+    ]
+    .map(senderReceiver => senderReceiver.transport)
+    .filter(dtlsTransport => dtlsTransport); // filter null/undefined
+
+    assert_greater_than_equal(dtlsTransports.length, 2, 'expect to find at least 2 RTCDtlsTransports');
```

Comment:
Yes the semantics kind of changed a bit, as I tried to avoid copy-pasting too much code when inlining. Though the original test don't really test that the transport objects in different `pc`s are different though, and validating that is a bit complicated and redundant.

I have simplified the code to just skip all the checkings. They just assume the browser is spec compliant and create a transport object after setting local description. The consequence is mainly error messages in tests is more obscure, for trying to access fields of a `null` or `undefined` object for current implementations.